deployment/terraform/output.go

```diff
@@ -87,11 +87,40 @@ type Output struct {
 
 // Instance is an AWS EC2 instance resource.
 type Instance struct {
-	PrivateIP  string `json:"private_ip"`
-	PublicIP   string `json:"public_ip"`
-	PublicDNS  string `json:"public_dns"`
-	PrivateDNS string `json:"private_dns"`
-	Tags       Tags   `json:"tags"`
+	PrivateIP      string `json:"private_ip"`
+	PublicIP       string `json:"public_ip"`
+	PublicDNS      string `json:"public_dns"`
+	PrivateDNS     string `json:"private_dns"`
+	Tags           Tags   `json:"tags"`
+	connectionType string
+}
+
+func (i *Instance) SetConnectionType(connType string) {
+	if connType != "private" && connType != "public" {
+		connType = ""
+	}
+	i.connectionType = connType
+}
+
+func (i Instance) GetConnectionType() string {
+	if i.connectionType == "" {
+		return "public"
+	}
+	return i.connectionType
+}
```

Comment:
Should we set `connectionType` to `public` in `SetConnectionType` instead of to the empty string, and then simply return what we have in `GetConnectionType`?

Reply:
If we end up in a code path that does not call `SetConnectionType` it will fail becasue it will be empty (as default string value). Made the change anyway because we should not end up in that situation anyways.